src/analyze/analyze.c

```diff
@@ -188,14 +200,28 @@ static void free_unit_times(struct unit_times *t, unsigned n) {
         free(t);
 }
 
+static void subtract_timestamp(usec_t *a, usec_t b) {
```

Comment:
Done.

BTW, is it correct to `assert(*a >= b)` here? Input does not come from the user, but it comes over the bus from a different daemon.